pkg/sql/sem/builtins/builtins.go

```diff
@@ -3671,6 +3699,32 @@ func stringToArray(str string, delimPtr *string, nullStr *string) (tree.Datum, e
 	return result, nil
 }
 
+// arrayToString implements the array_to_string builtin - arr is joined using
+// delim. If nullStr is non-nil, NULL values in the array will be replaced by
+// it.
+func arrayToString(arr *tree.DArray, delimPtr *string, nullStr *string) (tree.Datum, error) {
```

Comment:
I seem to recall we have very similar code to produce the string representation of arrays for pgwire. Since your new implementation here is more general, is there a way to use it in pgwire now?

Reply:
There's a few small behavior differences (when to quote the inner vars). I don't see a strong reason to DRY these, I think it's simpler and more Go-y to leave them as-is. I think combining them will be annoying if we ever start supporting nested arrays properly.